python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py

```diff
@@ -126,6 +131,57 @@ def test_no_match():
     assert len(res) == 0
 
 
+def test_match_below_and_abolve_deafault_query_threshold():
+    # Video1 contains five out of ten frames that match with indexed video4 -> query percent 50% and filtered
+    # Video2 contains eight out of ten frames that match with indexed video4 -> query percent 80% and return
+    # Video3 contains nine out of ten frames that match with indexed video4 -> query percent 90% and return
+    video1 = pdq_hashes_to_VPDQ_features(
+        random.sample(g1, 2) + random.sample(g2, 3) + random.sample(g3, 5)
+    )
+    video2 = pdq_hashes_to_VPDQ_features(
+        random.sample(g1, 1) + random.sample(g2, 1) + random.sample(g3, 8)
+    )
+    video3 = pdq_hashes_to_VPDQ_features(random.sample(g1, 1) + random.sample(g3, 9))
+    video4 = pdq_hashes_to_VPDQ_features(random.sample(g3, 10))
+
+    index = VPDQIndex.build([[vpdq_to_json(video4), VIDEO4_META_DATA]])
+    res = index.query(vpdq_to_json(video1))
+    assert len(res) == 0
+
+    res = index.query(vpdq_to_json(video2))
+    assert compare_match_result(res[0], VPDQIndexMatch(100, 80, 100, VIDEO4_META_DATA))
+
+    res = index.query(vpdq_to_json(video3))
```

Comment:
(ignorable): Many of these tests spend a lot of time converting between different formats of the same data. The index already unpacks these strings back into features. Exposing an inner function that takes features, or changing the accepted type of query to be either string or features (union type) would make these tests more straightforward.

Reply:
Will do this in a separate PR!